tmk_core/protocol/ps2_mouse.c

```diff
@@ -150,6 +150,13 @@ static inline void ps2_mouse_convert_report_to_hid(report_mouse_t *mouse_report)
     // remove sign and overflow flags
     mouse_report->buttons &= PS2_MOUSE_BTN_MASK;
 
+#ifdef PS2_MOUSE_INVERT_BUTTONS
+    int buttons = mouse_report->buttons;
+    buttons |= ((mouse_report->buttons >> PS2_MOUSE_BTN_LEFT) & 1U) << 1;
+    buttons |= (mouse_report->buttons >> PS2_MOUSE_BTN_RIGHT) & 1U;
+    mouse_report->buttons = buttons;
+#endif
```

Comment:
Is there any way to avoid setting `mouse_report->buttons` twice here?

Reply:
Sure.
https://github.com/qmk/qmk_firmware/pull/9205/commits/351b635b20e291f75be5df165e77efc62337d315

Reply:
That doesn't include the `&=` operation above.

Reply:
Sorry but I can't catch it. Where?

Reply:
Directly above this change.

Reply:
This one? `mouse_report->buttons &= PS2_MOUSE_BTN_MASK;`
Why I've to include it?

Reply:
MY question is whether we can assign `mouse_report->buttons` only once, even if `PS2_MOUSE_INVERT_BUTTONS` is defined.

Reply:
```
* Data format:
 * byte|     7          6          5          4       3           2          1        0
 * ----+----------------------------------------------------------------
 *    0|[Yovflw][Xovflw][Ysign ][Xsign ][ 1    ][Middle][Right ][Left  ]
```
This line exist to mask the overflow and sign.
I still can't catch what do you want to do.

Is possible but ugly to read.
```
    mouse_report->buttons = \
#ifdef PS2_MOUSE_INVERT_BUTTONS
    (mouse_report->buttons >> PS2_MOUSE_BTN_LEFT & 1U) << PS2_MOUSE_BTN_RIGHT | (mouse_report->buttons >> PS2_MOUSE_BTN_RIGHT & 1U) << PS2_MOUSE_BTN_LEFT | (mouse_report->buttons >> PS2_MOUSE_BTN_MIDDLE & 1U) << PS2_MOUSE_BTN_MIDDLE \
#else
    mouse_report->buttons \
#endif
    // remove sign and overflow flags
    & PS2_MOUSE_BTN_MASK;
```

Reply:
solved on https://github.com/qmk/qmk_firmware/pull/9205/commits/7811e9f2bedf281266c489da688fca773eb4e1cd

Reply:
I changed the logic to a LUT.
Looks easy to understand and more cleaner than use binary operators.